lib/file.coffee

```diff
@@ -67,3 +67,6 @@ class File
 
   isPathEqual: (pathToCompare) ->
     @path is pathToCompare or @realPath is pathToCompare
+
+  # Return a snapshot of the instance's filesystem properties for API consumption
+  getMetadata: -> {@symlink, @realPath, @status, @stats}
```

Comment:
> To be honest, I think we can probably justify having an extra filesystem call to return the missing info to stay consistent with tree-view's handling. Let's face it: there won't be nearly as many icon-displaying tabs as there will be icon-displaying file listings.

That's fair, although this feels like a clear indicator that this interface is inconsistent and more complex than it should need to be. I think we should try to design services in an elegant manner, so I am 👎 on passing different arguments based on the package, and I think we should issue `fs.stat` calls directly in the `FileIcons` service, delegating to it all the concerns about determining if a file is symlink and whatnot.

The only reasonable alternative I could think of (that doesn't involve a major redesign in Atom core) is to provide an alternative `iconClassForPath` method that returns a `Promise`, so that the UI thread never blocks and we can do whatever I/O is needed on that function. It would make the calling code a little more complex, but it's probably worth a try if performance-wise the file-icons package becomes a problem.